### ISSUE TITLE
An illegal reflective access operation has occurred

### ISSUE THREAD

**Issue by Miley1** [jruby/jruby#6118]
installed via RVM

```
$ uname -a 
#Linux linux-3v1j 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26) x86_64 GNU/Linux
$ java --version 
#openjdk 11.0.6 2020-01-14
OpenJDK Runtime Environment (build 11.0.6+10-post-Debian-1deb10u1)
OpenJDK 64-Bit Server VM (build 11.0.6+10-post-Debian-1deb10u1, mixed mode, sharing)
$ jruby --version 
#jruby 9.2.11.0 (2.5.7) 2020-03-02 612d7a05a6 OpenJDK 64-Bit Server VM 11.0.6+10-post-Debian-1deb10u1 on 11.0.6+10-post-Debian-1deb10u1 +jit [linux-x86_64
$irb
#WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules to method sun.nio.ch.NativeThread.signal(long)
WARNING: Please consider reporting this to the maintainers of com.headius.backport9.modules.Modules
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
$ jruby -J--illegal-access=debug -S irb
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules to method sun.nio.ch.NativeThread.signal(long)
	at com.headius.backport9.modules.Modules.checkOpen(Modules.java:58)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:40)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:30)
	at org.jruby.javasupport.Java.trySetAccessible(Java.java:1657)
	at org.jruby.RubyProcess.<clinit>(RubyProcess.java:939)
	at org.jruby.Ruby.<init>(Ruby.java:451)
	at org.jruby.Ruby.newInstance(Ruby.java:706)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
	at org.jruby.Main.main(Main.java:206)
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules to method sun.nio.ch.NativeThread.current()
	at com.headius.backport9.modules.Modules.checkOpen(Modules.java:58)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:40)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:30)
	at org.jruby.javasupport.Java.trySetAccessible(Java.java:1657)
	at org.jruby.RubyProcess.<clinit>(RubyProcess.java:939)
	at org.jruby.Ruby.<init>(Ruby.java:451)
	at org.jruby.Ruby.newInstance(Ruby.java:706)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
	at org.jruby.Main.main(Main.java:206)
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules to method java.lang.Object.finalize()
	at com.headius.backport9.modules.Modules.checkOpen(Modules.java:58)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:40)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:30)
	at org.jruby.javasupport.binding.MethodGatherer$PartitionedMethods.filterAccessible(MethodGatherer.java:674)
	at org.jruby.javasupport.binding.MethodGatherer$PartitionedMethods.<init>(MethodGatherer.java:654)
	at org.jruby.javasupport.binding.MethodGatherer$2.computeValue(MethodGatherer.java:266)
	at org.jruby.javasupport.binding.MethodGatherer$2.computeValue(MethodGatherer.java:263)
	at java.base/java.lang.ClassValue.getFromHashMap(ClassValue.java:226)
	at java.base/java.lang.ClassValue.getFromBackup(ClassValue.java:208)
	at java.base/java.lang.ClassValue.get(ClassValue.java:114)
	at org.jruby.javasupport.binding.MethodGatherer.eachAccessibleMethod(MethodGatherer.java:176)
	at org.jruby.javasupport.binding.MethodGatherer.getMethods(MethodGatherer.java:156)
	at org.jruby.javasupport.binding.MethodGatherer.setupMethods(MethodGatherer.java:558)
	at org.jruby.javasupport.binding.MethodGatherer.initialize(MethodGatherer.java:127)
	at org.jruby.javasupport.binding.ClassInitializer.initialize(ClassInitializer.java:63)
	at org.jruby.javasupport.binding.Initializer.setupProxyClass(Initializer.java:54)
	at org.jruby.javasupport.Java.createProxyClass(Java.java:549)
	at org.jruby.javasupport.Java.generateClassProxy(Java.java:498)
	at org.jruby.javasupport.Java.createProxyClassForClass(Java.java:463)
	at org.jruby.javasupport.JavaSupportImpl$3.computeValue(JavaSupportImpl.java:127)
	at org.jruby.javasupport.JavaSupportImpl$3.computeValue(JavaSupportImpl.java:119)
	at org.jruby.util.collections.MapBasedClassValue.get(MapBasedClassValue.java:25)
	at org.jruby.javasupport.JavaSupportImpl.getProxyClassFromCache(JavaSupportImpl.java:202)
	at org.jruby.javasupport.Java.getProxyClass(Java.java:430)
	at org.jruby.javasupport.JavaClass.getProxyClass(JavaClass.java:104)
	at org.jruby.javasupport.Java.load(Java.java:128)
	at org.jruby.Ruby.initJavaSupport(Ruby.java:1734)
	at org.jruby.Ruby.initBootLibraries(Ruby.java:545)
	at org.jruby.Ruby.<init>(Ruby.java:514)
	at org.jruby.Ruby.newInstance(Ruby.java:706)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules to method java.lang.Object.clone()
	at com.headius.backport9.modules.Modules.checkOpen(Modules.java:58)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:40)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:30)
	at org.jruby.javasupport.binding.MethodGatherer$PartitionedMethods.filterAccessible(MethodGatherer.java:674)
	at org.jruby.javasupport.binding.MethodGatherer$PartitionedMethods.<init>(MethodGatherer.java:654)
	at org.jruby.javasupport.binding.MethodGatherer$2.computeValue(MethodGatherer.java:266)
	at org.jruby.javasupport.binding.MethodGatherer$2.computeValue(MethodGatherer.java:263)
	at java.base/java.lang.ClassValue.getFromHashMap(ClassValue.java:226)
	at java.base/java.lang.ClassValue.getFromBackup(ClassValue.java:208)
	at java.base/java.lang.ClassValue.get(ClassValue.java:114)
	at org.jruby.javasupport.binding.MethodGatherer.eachAccessibleMethod(MethodGatherer.java:176)
	at org.jruby.javasupport.binding.MethodGatherer.getMethods(MethodGatherer.java:156)
	at org.jruby.javasupport.binding.MethodGatherer.setupMethods(MethodGatherer.java:558)
	at org.jruby.javasupport.binding.MethodGatherer.initialize(MethodGatherer.java:127)
	at org.jruby.javasupport.binding.ClassInitializer.initialize(ClassInitializer.java:63)
	at org.jruby.javasupport.binding.Initializer.setupProxyClass(Initializer.java:54)
	at org.jruby.javasupport.Java.createProxyClass(Java.java:549)
	at org.jruby.javasupport.Java.generateClassProxy(Java.java:498)
	at org.jruby.javasupport.Java.createProxyClassForClass(Java.java:463)
	at org.jruby.javasupport.JavaSupportImpl$3.computeValue(JavaSupportImpl.java:127)
	at org.jruby.javasupport.JavaSupportImpl$3.computeValue(JavaSupportImpl.java:119)
	at org.jruby.util.collections.MapBasedClassValue.get(MapBasedClassValue.java:25)
	at org.jruby.javasupport.JavaSupportImpl.getProxyClassFromCache(JavaSupportImpl.java:202)
	at org.jruby.javasupport.Java.getProxyClass(Java.java:430)
	at org.jruby.javasupport.JavaClass.getProxyClass(JavaClass.java:104)
	at org.jruby.javasupport.Java.load(Java.java:128)
	at org.jruby.Ruby.initJavaSupport(Ruby.java:1734)
	at org.jruby.Ruby.initBootLibraries(Ruby.java:545)
	at org.jruby.Ruby.<init>(Ruby.java:514)
	at org.jruby.Ruby.newInstance(Ruby.java:706)
	at org.jruby.Main.internalRun(Main.java:273)
	at org.jruby.Main.run(Main.java:234)
WARNING: Illegal reflective access by com.headius.backport9.modules.Modules to field java.io.FileDescriptor.fd
	at com.headius.backport9.modules.Modules.checkOpen(Modules.java:58)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:40)
	at com.headius.backport9.modules.Modules.trySetAccessible(Modules.java:30)
	at org.jruby.javasupport.Java.trySetAccessible(Java.java:1657)
	at org.jruby.util.io.FilenoUtil$ReflectiveAccess.<clinit>(FilenoUtil.java:236)
	at org.jruby.util.io.FilenoUtil.getFilenoUsingReflection(FilenoUtil.java:123)
	at org.jruby.util.io.FilenoUtil.filenoFrom(FilenoUtil.java:119)
	at org.jruby.util.io.ChannelFD.initFileno(ChannelFD.java:47)
	at org.jruby.util.io.ChannelFD.<init>(ChannelFD.java:33)
	at org.jruby.util.io.ChannelFD.<init>(ChannelFD.java:43)
	at org.jruby.RubyIO.<init>(RubyIO.java:174)
	at org.jruby.RubyFile.<init>(RubyFile.java:279)
	at org.jruby.parser.Parser.parse(Parser.java:100)
	at org.jruby.Ruby.parseFileAndGetAST(Ruby.java:2527)
	at org.jruby.Ruby.parseFileFromMainAndGetAST(Ruby.java:2520)
	at org.jruby.Ruby.parseFileFromMain(Ruby.java:2504)
	at org.jruby.Ruby.parseFromMain(Ruby.java:991)
	at org.jruby.Ruby.runFromMain(Ruby.java:937)
	at org.jruby.Main.doRunFromMain(Main.java:412)
	at org.jruby.Main.internalRun(Main.java:304)
	at org.jruby.Main.run(Main.java:234)
	at org.jruby.Main.main(Main.java:206)
jruby-9.2.11.0 :001 > 
```

**Comment by headius** [comment]
I am unable to reproduce this with a combination of JRuby 9.2.11.0 (installed via rvm) and jruby-launcher 1.1.12 (released last week). Can you confirm you still get the warnings with this combination?

```
[] ~/projects/jruby $ java -version
openjdk version "11.0.4" 2019-07-16
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.4+11)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.4+11, mixed mode)

[] ~/projects/jruby $ which ruby
/Users/headius/.rvm/rubies/jruby-9.2.11.0/bin/ruby

[] ~/projects/jruby $ file `which ruby`
/Users/headius/.rvm/rubies/jruby-9.2.11.0/bin/ruby: Mach-O 64-bit executable x86_64

[] ~/projects/jruby $ ruby -v
jruby 9.2.11.0 (2.5.7) 2020-03-02 612d7a05a6 OpenJDK 64-Bit Server VM 11.0.4+11 on 11.0.4+11 +jit [darwin-x86_64]

[] ~/projects/jruby $ ruby -Xversion
JRuby Launcher Version 1.1.12

[] ~/projects/jruby $ irb
jruby-9.2.11.0 :001 > exit

[] ~/projects/jruby $ jruby -J--illegal-access=debug -S irb
jruby-9.2.11.0 :001 > exit
```

**Comment by headius** [comment]
I believe this is the same cause as #6049 and should be fixed by the updated jruby-launcher in 1.1.13.

Install the latest jruby-launcher gem or force rvm to reinstall.
